src/oid.rs

```diff
@@ -168,7 +168,7 @@ impl ObjectId {
         // Static mutexes are currently unstable; once they have been
```

Comment:
The counter is stored as a `usize`, so a second "overflow event" will occur when that reaches its max. Also, from looking at the implementation, once the counter reaches usize's max, it will be randomly generated again and not overflow properly. To verify this, I'd add another test case that sets the counter to `usize`'s max and see what happens. It should fail I think, and so `gen_count` will need to be updated to fix it.

Reply:
makes sense. fixed by adding a boolean variable and making sure counter resets to 0 in case of usize_max overflow (instead of random generation)

Reply:
The atomic boolean approach is susceptible to race conditions. In order to maintain consistency between the two different variables you'd need a lock. Unfortunately, we can't use locks either (as the comment in the code suggests) since it would needlessly inhibit performance and prevent `ObjectId::new` from being used in non-blocking contexts.

To do this without locks, we could use `lazy_static` to ensure the random generation only happens the first time, and all subsequent rollovers will just happen normally via addition without the usage of a separate atomic variable. As part of that, you can just remove the comment about adding locks in the future.